Mathlib/MeasureTheory/Measure/LevyProkhorovMetric.lean

```diff
@@ -387,19 +435,220 @@ lemma continuous_levyProkhorov_to_probabilityMeasure :
+  -- at each point `P : ProbabilityMeasure Ω`.
+  rw [continuous_iff_continuousAt]
+  intro P
+
+  -- To check continuity, fix `ε > 0`. To leave some wiggle room, be ready to use `ε/3 > 0` instead.
+  rw [continuousAt_iff']
+  intro ε ε_pos
+  have third_ε_pos : 0 < ε / 3 := by linarith
+  have third_ε_pos' : 0 < ENNReal.ofReal (ε / 3) := ofReal_pos.mpr third_ε_pos
+
+  -- First use separability to choose a countable partition of `Ω` into measurable
+  -- subsets `Es n ⊆ Ω` of small diamater, `diam (Es n) < ε/3`.
+  obtain ⟨Es, Es_mble, Es_bdd, Es_diam, Es_cover, Es_disjoint⟩ :=
+    SeparableSpace.exists_measurable_partition_diam_le Ω third_ε_pos
+
+  -- Instead of the whole space `Ω = ⋃ n ∈ ℕ, Es n`, focus on a large but finite
+  -- union `⋃ n < N, Es n`, chosen in such a way that the complement has small `P`-mass,
+  -- `P (⋃ n < N, Es n)ᶜ < ε/3`.
+  have exhaust :=
+    @tendsto_measure_biUnion_Ici_zero_of_pairwise_disjoint Ω _ P.toMeasure _ Es Es_mble Es_disjoint
+  simp only [tendsto_atTop_nhds, Function.comp_apply] at exhaust
+  obtain ⟨N, hN⟩ := exhaust (Iio (ENNReal.ofReal (ε / 3))) third_ε_pos' isOpen_Iio
+  specialize hN N le_rfl
+
+  -- With the finite `N` fixed above, consider the finite collection of open sets of the form
+  -- `Gs J = thickening (ε/3) (⋃ j ∈ J, Es j)`, where `J ⊆ {0, 1, ..., N-1}`.
+  have Js_finite : Set.Finite {J | J ⊆ Iio N} := Finite.finite_subsets <| finite_Iio N
+  set Gs := (fun (J : Set ℕ) ↦ thickening (ε/3) (⋃ j ∈ J, Es j)) '' {J | J ⊆ Iio N}
+  have Gs_open : ∀ (J : Set ℕ), IsOpen (thickening (ε/3) (⋃ j ∈ J, Es j)) :=
+    fun J ↦ isOpen_thickening
+
+  -- Any open set `G ⊆ Ω` determines a neighborhood of `P` consisting of those `Q` that
+  -- satisfy `P G < Q G + ε/3`.
+  have mem_nhds_P :
+      ∀ G, IsOpen G → ({Q | P.toMeasure G < Q.toMeasure G + ENNReal.ofReal (ε/3)} ∈ 𝓝 P) :=
+    fun G G_open ↦ P.toMeasure_add_pos_gt_mem_nhds G_open third_ε_pos' ofReal_ne_top
+
+  -- Assume that `Q` is in the neighborhood of `P` such that for each `J ⊆ {0, 1, ..., N-1}`
+  -- we have `P (Gs J) < Q (Gs J) + ε/3`.
+  filter_upwards [(Finset.iInter_mem_sets Js_finite.toFinset).mpr <|
+                    fun J _ ↦ mem_nhds_P _ (Gs_open J)] with Q hQ
+  simp only [Finite.mem_toFinset, mem_setOf_eq, thickening_iUnion, mem_iInter] at hQ
+
+  -- Note that in order to show that the Lévy-Prokhorov distance `LPdist P Q` is small (`≤ 2*ε/3`),
+  -- it suffices to show that for arbitrary subsets `B ⊆ Ω`, the measure `P B` is bounded above up
+  -- to a small error by the `Q`-measure of a small thickening of `B`.
+  apply lt_of_le_of_lt ?_ (show 2*(ε/3) < ε by linarith)
+  rw [LevyProkhorov.dist_def, levyProkhorovDist, levyProkhorovEDist_comm]
+  apply levyProkhorovDist_le_of_forall_le
+  · linarith
+  · -- Fix an arbitrary set `B ⊆ Ω`, and an arbitrary `δ > 2*ε/3` to gain some room for error
+    -- and for thickening.
+    intro δ B δ_gt _
```

Comment:
```suggestion
  -- Fix an arbitrary set `B ⊆ Ω`, and an arbitrary `δ > 2*ε/3` to gain some room for error
  -- and for thickening.
  apply levyProkhorovDist_le_of_forall_le _ _ (by linarith) (fun δ B δ_gt _ ↦ ?_)
```
You can put the linarith and the intro step in the `apply`. The main advantage is that you can unindent the rest of your proof.